### PR TITLE
Isaac Del Rosario -- Carets

### PR DESCRIPTION
**Pull request by hisaacdelr**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I used the Edamam API documentation as a reference to the parameters the API can take. I then tested for base cases like how it would act with one ingredient or two or more ingredients. Then the edge cases where if it can take an empty query or a gibberish query.  |
| Describe your API Wrapper.  How did you decide on the methods you created? | I created an initial search method that would look for the recipes given a query. When thinking about how to call the recipe needed for the show page, I initially tried to create a separate method that would only look for one specific recipe given the URI. But then realized my original function can look up recipes with the URI as well so I recycled my method to be used for both searching multiple recipes given a query like "chicken" AND a specific recipe given its URI |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I checked if the user passes in a gibberish query and also a blank query. With how Edamam's API works, the user will always be given a blank array at the least if their query is invalid OR blank (given the same request path used in my APIWrapperMethod). Implementation-wise, it was up to my discretion on what I wanted to do with the blank arrays in my controller. But left alone, either one of these calls could have led to a 200 status so implementation based on these tests were important.|
| Explain how VCR aids in testing an API. | VCR helped reduce the amount of API calls I had to make to the service when running my tests which helped because with too many calls, I could get timed out. It also accounts for the case when the API service does time out, and using VCR allowed me to reuse the responses of old API calls it has already made. |
| What is the Heroku URL of your deployed application? | http://api-muncher-c8.herokuapp.com/ |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require "httparty"
+require "awesome_print"
+
+class EdamamApiWrapper
+  # Your code here!
+  BASE_URL = "https://api.edamam.com/"
+  APP_KEY = ENV["EDAMAM_KEY"]
+  APP_ID = ENV["EDAMAM_ID"]
+
+  def self.search(query, app_key = nil, app_id = nil)
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:10]: No method to get an individual recipe?

File: test/lib/edamam_api_wrapper_test.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      recipes.must_be_instance_of Array
+      recipes.must_equal []
+
+      EdamamApiWrapper.search("").must_be_instance_of Array
+    end
+  end
+
+  it "Will not break if invalid API tokens are passed" do
+    VCR.use_cassette("invalid_API_key") do
+      recipes = EdamamApiWrapper.search("chicken", "poop", "poop")
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:52]: 💩

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commits and good commit messages. |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check  |
| Semantic HTML | A few too many `div` element in the layout file, otherwise fine. |
| Errors are reported to the user | Nope |
| API Wrapper to handle the API requests | Check, but it' missing a way to get an individual recipe  |
| Controller testing | Check |
| Lib testing | :hankey:, check |
| Search Functionality | Check |
| List Functionality | Yep |
| Show individual item functionality (link to original recipe opens in new tab) | Not in a separate tab, but otherwise ok. |
| **Styling** |  |
| Foundation Styling for responsive layout | Well done |
| List View shows 10 items at a time/pagination | Good pagination |
| The app is styled to create an attractive user interface | Nicely done here |
| **API Features** | |
| The App attributes Edaman | Yep |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Check |
|  **Overall** | Very nicely done.  Great work and nice styling.   |
